system/lib/wasmfs/syscalls.cpp

```diff
@@ -1104,4 +1112,66 @@ long __syscall_ftruncate64(long fd, long low, long high) {
   }
   return ret;
 }
+
+long __syscall_pipe(long fd) {
+  auto* fds = (__wasi_fd_t*)fd;
+
+  // A global singleton PipeBackend is used for all pipes.
+  static auto backend = wasmFS.addBackend(std::make_unique<PipeBackend>());
+
```

Comment:
I don't think we actually need to create a `PipeBackend` or even define the class. As a precedent, stdin and friends do not have a backend either.

Reply:
Oh, thanks, I didn't realize a null backend was ok. I'll remove the backend and also update the comment on the backend field on File to indicate it is nullable.